app/controllers/saml_idp_controller.rb

```diff
@@ -36,8 +36,9 @@ def logout
 
     # Plumb the fingerprint through to the internal service_provider representation
     if saml_request&.service_provider
-      saml_request.service_provider.fingerprint =
-         Fingerprinter.fingerprint_cert(matching_cert || current_service_provider.ssl_certs.first)
+      saml_request.service_provider.fingerprint = matching_cert ?
+        Fingerprinter.fingerprint_cert(matching_cert) :
+        'some-non-nil-value'
```

Comment:
I don't fully understand the library, but to avoid the magic string, could we do

```ruby
if matching_cert
  saml_request.service_provider.fingerprint = Fingerprinter.fingerprint_cert(matching_cert)
end
```

Reply:
the library gets cranky if `.fingerprint` is nil, so that's what I'm trying to avoid at all costs

Reply:
I guess my confusion is we weren't setting fingerprint before, so it was getting set somewhere? If we need to override we can with the `if`, and if there's no matching cert, it feels like we should error earlier in the request.

Reply:
We used to pass `fingerprint` every time: https://github.com/18F/identity-idp/pull/4851/files#diff-fdfe3f731c3c1bb52752814efbd9b353dec85fca38e11f0ccc05e8622b9893beL29-L31

But since fingerprint is tied to a specific cert, we can't do that for multiple certs

Reply:
> My thinking was that if `cert` is `nil`, `Array(cert)` results in `[nil]`, whereas `Array(cert).compact` would result in `[]`.

```ruby
irb(main):008:0> Array(nil)
=> []
```